### PR TITLE
chore: refactor button v1 usages to v2

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Update old implementations of buttons to v2.
- Introduce a new variant `Subtle`. Currently used on Search suggestion list and profile social links.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-113 #done

### CODE REVIEW

File: packages/shared/src/components/search/SearchBarSuggestion.tsx [hunk 1]
```diff
@@ -73,16 +74,13 @@ export const SearchBarSuggestion = ({
...
-        'btn-secondary !h-auto min-h-[2.5rem] w-fit border-theme-divider-tertiary text-theme-label-tertiary typo-subhead',
-        className,
-      )}
+      variant={ButtonVariant.Subtle}
+      className={classNames('w-fit !justify-start typo-subhead', className)}
       onClick={handleSuggestionsClick}
       {...props}
-    />
+    >
+      <span className="w-fit tablet:!line-clamp-1">{children}</span>
```
Comment (sshanzel) [inline packages/shared/src/components/search/SearchBarSuggestion.tsx:83]: Tested existing implementations if the ellipsis still works as intended.

https://preview.app.daily.dev/search/chat

Screenshot 2024-02-19 at 9 35 31 PM

File: packages/shared/src/components/post/NewComment.tsx [hunk 2]
```diff
@@ -130,10 +130,10 @@ function NewCommentComponent(
...
       </span>
       <Button
-        buttonSize={buttonSize[size]}
-        className="btn-secondary ml-auto"
-        readOnly
-        tag="span"
+        size={buttonSize[size]}
+        className="ml-auto text-theme-label-primary"
+        variant={ButtonVariant.Secondary}
+        disabled
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/post/NewComment.tsx:136]: This is actually not clickable, this is a nested button that does the action, which is why previously it had a `readOnly` property.

[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/post/NewComment.tsx:136]: This is the button in subject. The "Post" button. We should probably not use the button component, but not sure what's best.

Screenshot 2024-02-19 at 10 58 40 PM

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/post/NewComment.tsx:136]: Yeah probably should be some other component realistically.
But you ensured it still works fully as intended? (open the comment part)

If so no blocker from my side.

**Review by rebelchris — APPROVED** [review]

File: packages/shared/src/components/post/NewComment.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/post/NewComment.tsx:136]: Yep. This is currently deployed on preview.

https://preview.app.daily.dev/posts/5JU4K2hMM

**Comment by sshanzel** [comment]
@kopancek would love to get your review as well before merging.

File: packages/storybook/stories/ButtonV2.stories.tsx [hunk 3]
```diff
@@ -47,7 +47,7 @@ export const Sizes: Story = {
             </Button>
           </span>
           <span key={size + '_iconOnly'}>
-            <Button {...props} size={size} icon={<ShareIcon />} Tag="span" />
+            <Button {...props} size={size} icon={<ShareIcon />} />
```
Comment (nensidosari) [inline packages/storybook/stories/ButtonV2.stories.tsx:50]: Why is Tag span removed here?

**Review by nensidosari — APPROVED** [review]
Looks great, one suggestion only 🎉

File: packages/storybook/stories/ButtonV2.stories.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/storybook/stories/ButtonV2.stories.tsx:50]: It is not really an accepted prop. It only accepts `a` or `button`.

File: packages/shared/src/components/post/NewComment.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ import {
   ProfileImageSize,
   ProfilePicture,
 } from '../ProfilePicture';
-import { Button, ButtonSize } from '../buttons/Button';
+import { Button, ButtonSize, ButtonVariant } from '../buttons/ButtonV2';
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/post/NewComment.tsx:17]: Yeah, was thinking of that too, but then the diff history could've been harder to look at. Maybe on a separate PR or before I merge.

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/post/NewComment.tsx:17]: Yeah let's do that indeed :) 
Sounds ok and will avoid confusion :)

File: packages/storybook/stories/ButtonV2.stories.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/storybook/stories/ButtonV2.stories.tsx:50]: Lol, yeah. TS would go crazy until it gets the correct setup. We can maybe have it as an improvement in the future, part of V3 as you mentioned 👀

**Comment by sshanzel** [comment]
It seems that we have no pending actionable item, in that case, I will start the removal of the v2 on the name.

File: packages/shared/src/components/buttons/Button.tsx [hunk 5]
```diff
@@ -6,118 +6,111 @@ import React, {
   Ref,
```
Comment (sshanzel) [inline packages/shared/src/components/buttons/Button.tsx:6]: Nothing really changed much here, this was just the ButtonV2.

**Comment by sshanzel** [comment]
@dailydotdev/web-team not sure if anyone would be interested, but the build and tests passed, though feel free to review if you want to.

**Review by kopancek — APPROVED** [review]
Looks good to me, checked the rename commit as well 👍

**Pull request merged**